### ISSUE TITLE
Token amount checks for giveaway endpoint

### ISSUE THREAD

**Issue by juliancwirko** [juliancwirko/elven-nft-minter-sc#27]
- giveaway endpoint is only for the owner, but it doesn't respect the limits per address, so it needs some checks and improvements for big amounts in one transaction because it will reach the max gas limit per one transaction - it should be probably also improved for the standard mint endpoint. Something around 55 tokens per one transaction should be ok. So it will likely be never reached because most of the collections will add lower limits than that. But it should adequately handle that.

**Comment by butavpaul** [comment]
In my opinion, should be limited only / transactions and not / wallets limit. The owner can keep NFTs for air drops, giveaways/contests. Some of the projects are keeping even 200 nfts.
And also even if it's max nft/wallet it's owner decision to giveaway more.

**Comment by juliancwirko** [comment]
Good point. I will think about how to unify it. Also, some updates in the CLI will probably be required.

**Comment by juliancwirko** [comment]
For now, on the development branch, I removed the limit, and only the CLI will keep them when triggering the 'mint' or 'giveaway'. I also think I won't implement any checks for that on the SC because it is not natural. The 'UI' app (CLI or dapp) should keep that or someone who tries to use the SC directly.

**Comment by juliancwirko** [comment]
published
